src/Raven.Studio/typescript/viewmodels/database/tasks/editReplicationHubTask.ts

```diff
@@ -0,0 +1,480 @@
+                        .done((accessResult: Raven.Client.Documents.Operations.Replication.ReplicationHubAccessResult) => {
+                            self.processResults(accessResult);
+                            deferredAccessInfo.resolve();
+                        })
+                        .fail(() => {
+                            deferredAccessInfo.reject();
+                            router.navigate(appUrl.forOngoingTasks(this.activeDatabase()));
+                        });
+                })
+                .fail(() => {
+                    deferredHubTaskInfo.reject();
+                    router.navigate(appUrl.forOngoingTasks(this.activeDatabase()));
+                });
+            
+            
+        } else {
+            // 2. Creating a new task
+            this.isNewTask(true);
+            this.editedHubTask(ongoingTaskReplicationHubEditModel.empty());
+            deferredHubTaskInfo.resolve();
+        }
+
+        deferredHubTaskInfo.done(() => this.initObservables());
+
+        if (args.taskId) {
+            return $.when<any>(this.loadPossibleMentors(), deferredHubTaskInfo, deferredAccessInfo); 
+        }
+        
+        return $.when<any>(this.loadPossibleMentors(), deferredHubTaskInfo);
+    }
+
+    private processResults(accessResult: Raven.Client.Documents.Operations.Replication.ReplicationHubAccessResult) {
+        const accessItems = accessResult.Results.map(x => {
+            
+            const certificate = new replicationCertificateModel(x.Certificate);
+            const h2sPaths = x.AllowedHubToSinkPaths.map(x => new prefixPathModel(x));
+            const s2hPaths = x.AllowedSinkToHubPaths.map(x => new prefixPathModel(x));
+           
+            return new replicationAccessHubModel(x.Name, certificate, h2sPaths, s2hPaths, this.editedHubTask().withFiltering() ,false);
+        });
+
+        this.editedHubTask().replicationAccessItems(accessItems);
+    }
+    
+    attached() {
+        super.attached();
+
+        popoverUtils.longWithHover($("#replication-filtering-info"),
+            {
+                content:
+                    "<ul class='margin-bottom margin-bottom-xs'>" +
+                        "<li><small>Check this toggle in order to be able to define filtering on this Replication Hub task.</small></li>" +
+                        "<li><small>You will be able to define the replication filtering when editing the task after saving this configuration.</small></li>" +
+                        "<li><small class='text-warning'><i class='icon-warning'></i>" +
+                            "<span>Note: This parameter cannot be modified after saving this configuration.</span></small>" +
+                        "</li>" +
+                    "</ul>"
+            });
+    }
+    
+    private loadPossibleMentors() {
+        return new getPossibleMentorsCommand(this.activeDatabase().name)
+            .execute()
+            .done(mentors => this.possibleMentors(mentors));
```

Comment:
I wonder if we want to handle failure here?

Reply:
done - handled fail inside of `getPossibleMentorsCommand`